src/core/dbus-execute.c

```diff
@@ -968,6 +1050,9 @@ int bus_set_transient_exec_command(
                         _cleanup_free_ char *a = NULL, *t = NULL;
```

Comment:
I'm not sure if this is intended, but I think this loop writes out the `Exec` properties too many times. For example `systemctl cat <transient unit>` on 
```
$ systemd-run -p "ExecStart=/bin/echo first" -p "ExecStart=/bin/echo second" -p "ExecStart=/bin/echo third" -p "Type=oneshot" /bin/sleep infinity
```
resulted in 

```
# /run/systemd/transient/run-u19343.service
# This is a transient unit file, created programmatically via the systemd API. Do not edit.
[Unit]
Description=/bin/sleep infinity

[Service]
ExecStart=
ExecStart=@/bin/echo "/bin/echo" "first"
ExecStart=
ExecStart=@/bin/echo "/bin/echo" "first"
ExecStart=@/bin/echo "/bin/echo" "second"
ExecStart=
ExecStart=@/bin/echo "/bin/echo" "first"
ExecStart=@/bin/echo "/bin/echo" "second"
ExecStart=@/bin/echo "/bin/echo" "third"
Type=oneshot
ExecStart=
ExecStart=@/bin/echo "/bin/echo" "first"
ExecStart=@/bin/echo "/bin/echo" "second"
ExecStart=@/bin/echo "/bin/echo" "third"
ExecStart=@/bin/sleep "/bin/sleep" "infinity"
```